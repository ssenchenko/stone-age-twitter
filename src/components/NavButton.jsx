import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ButtonStyled = styled.button`
  border: none;
  outline: none;
  background: #ffffff;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  &:hover {
    color: ${({ theme }) => (theme.activeColor)};
  }
`;

const NavButton = ({ name, onClickFunc }) => (
  <ButtonStyled onClick={onClickFunc}>{name}</ButtonStyled>
);

NavButton.propTypes = {
  name: PropTypes.string.isRequired,
  onClickFunc: PropTypes.func.isRequired,
};

export default NavButton;
