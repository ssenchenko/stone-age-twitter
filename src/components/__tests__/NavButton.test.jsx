import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import DefaultTheme from '../../settings/themes';
import NavButton from '../NavButton';

const onClick = jest.fn();

describe('NavButton', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={DefaultTheme}>
          <NavButton name="Test" onClickFunc={onClick} />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
