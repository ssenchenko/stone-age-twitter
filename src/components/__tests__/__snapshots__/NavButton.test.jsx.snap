// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton renders correctly 1`] = `
.c0 {
  border: none;
  outline: none;
  background: #ffffff;
  padding: 15px 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}

.c0:hover {
  color: lightgreen;
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  Test
</button>
`;
